CSharpHacks/CSharpHacks/UriHacks.cs

```diff
@@ -1,22 +1,81 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
 
 namespace CSharpHacks
 {
     public static class UriHacks
     {
-        public static Uri AddParameter(this Uri url, string name, string value)
+        public static Uri AddParameter(this Uri url, IEnumerable<KeyValuePair<string, string>> parameters)
+            => _appendToQuery(url, parameters);
+
+        public static Uri AddParameter(this Uri url, params KeyValuePair<string, string>[] parameters)
+            => url.AddParameter(parameters as IEnumerable<KeyValuePair<string, string>>);
+
+        public static Uri AddParameter(this Uri url, string key, string value = null)
+            => url.AddParameter(new KeyValuePair<string, string>(key, value));
+
+
+        private static Uri _appendToQuery(Uri url, IEnumerable<KeyValuePair<string, string>> parameters)
         {
+            //Create parameter string builder
```

Comment:
couldn't this potentially be done in a more concise manner with a string.join and a Where clause to remove empty keys?

Reply:
Good idea. I think I had a case of premature optimization, no reason to shave off those 2 ms in the tests.
I changed it to use Linq now.